pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -659,6 +659,40 @@ class Build {
         }
     }
 
+    def waitForANodeToBecomeActive(def label) {
+        def NodeHelper = context.library(identifier: 'openjdk-jenkins-helper@master').NodeHelper
+
+        if (NodeHelper.nodeIsOnline(label)) {
+            return
+        }
+
+        context.println("No active node matches this label: " + label)
+
+        int activeNodeTimeout = 0
+        if (buildConfig.ACTIVE_NODE_TIMEOUT.isInteger()) {
+            activeNodeTimeout = buildConfig.ACTIVE_NODE_TIMEOUT as Integer
+        }
+
+
+        if (activeNodeTimeout > 0) {
+        context.println("Will check again periodically until either one comes online, or " + buildConfig.ACTIVE_NODE_TIMEOUT + " minutes (ACTIVE_NODE_TIMEOUT) has passed.")
+            int x = 0
+            while (x < activeNodeTimeout) {
+                context.sleep(time: activeNodeTimeout, unit: "MINUTES")
+                if (NodeHelper.nodeIsOnline(label)) {
+                    context.println("A node which matches this label is now active: " + label)
+                    return
+                }
+                x++
```

Comment:
This loop doesn't look right, the context.sleep() will sleep for activeNodeTimeout "MINUTES" each time through the loop until x == activeNodeTimeout. So if activeNodeTimeout is 5 minutes it is actually going to sleep for 5x5=25 mins ?

Reply:
Ah, good spot. Fixed.